iterator.go

```diff
@@ -0,0 +1,63 @@
+package redis
+
+import "sync"
+
+// ScanIterator is used to incrementally iterate over a collection of elements.
+// Instances are thread-safe and can be used across multiple goroutines.
+type ScanIterator struct {
+	scn *Scanner
+	pos int
+	mu  sync.Mutex
+}
+
+// Err returns the last iterator error, if any
+func (i *ScanIterator) Err() error {
+	i.mu.Lock()
+	err := i.scn.Err()
+	i.mu.Unlock()
+	return err
+}
+
+// Next advances the cursor and returns true if more values can be read
+func (i *ScanIterator) Next() bool {
+	i.mu.Lock()
+	defer i.mu.Unlock()
+
+	// Instantly return on errors
+	if i.scn.Err() != nil {
+		return false
+	}
+
+	// Advance cursor, check if we are still within range
+	if i.pos < len(i.scn.page) {
+		i.pos++
+		return true
+	}
+
+	// Return if there is more data to fetch
+	if i.scn.cursor == 0 {
+		return false
+	}
+
+	// Fetch next page
+	i.scn._args[1] = i.scn.cursor
+	i.scn.reset()
+	i.scn.c.Process(i.scn.ScanCmd)
```

Comment:
There is also opportunity to improve var names, e.g.:
- it.cmd._args
- it.client.Process